docker/kubernetes-agent-tentacle/scripts/configure-and-run.sh

```diff
@@ -246,15 +261,62 @@ function registerTentacle() {
       '--password' "$ServerPassword")
   fi
 
-  if [[ -n "$TargetName" ]]; then
-    ARGS+=('--name' "$TargetName")
+  tentacle "${ARGS[@]}"
+}
+
+function getTentacleAsDeploymentTargetRegistrationArgs() {
+  local ARGS=()
+
+  if [[ -n "$TargetEnvironment" ]]; then
+    IFS=',' read -ra ENVIRONMENTS <<<"$TargetEnvironment"
+    for i in "${ENVIRONMENTS[@]}"; do
+      ARGS+=('--environment' "$i")
+    done
+  fi
+
+  if [[ -n "$TargetRole" ]]; then
+    IFS=',' read -ra ROLES <<<"$TargetRole"
+    for i in "${ROLES[@]}"; do
+      ARGS+=('--role' "$i")
+    done
+  fi
+
+  if [[ -n "$TargetTenant" ]]; then
+    IFS=',' read -ra TENANTS <<<"$TargetTenant"
+    for i in "${TENANTS[@]}"; do
+      ARGS+=('--tenant' "$i")
+    done
+  fi
+
+  if [[ -n "$TargetTenantTag" ]]; then
+    IFS=',' read -ra TENANTTAGS <<<"$TargetTenantTag"
+    for i in "${TENANTTAGS[@]}"; do
+      ARGS+=('--tenanttag' "$i")
+    done
   fi
 
   if [[ -n "$TargetTenantedDeploymentParticipation" ]]; then
     ARGS+=('--tenanted-deployment-participation' "$TargetTenantedDeploymentParticipation")
   fi
 
-  tentacle "${ARGS[@]}"
+  if [[ -n "$DefaultNamespace" ]]; then
+    ARGS+=('--default-namespace' "$DefaultNamespace")
+  fi
+
+  echo "${ARGS[@]}"
+}
+
+function getTentacleAsWorkerRegistrationArgs() {
+  local ARGS=()
+
+  if [[ -n "$WorkerPools" ]]; then
+    IFS=',' read -ra WORKERPOOLS <<<"$WorkerPools"
+    for i in "${WORKERPOOLS[@]}"; do
+      ARGS+=('--workerpool' "$i")
+    done
+  fi
+
+  echo "${ARGS[@]}"
 }
```

Comment:
Nit: This is an impressive bash script. I wonder if it can be pulled apart so that it may be tested and make future changes to it easier.

Reply:
Yea, agreed that this bash script has gotten a bit big. I think it'd be ideal to have less logic in bash and more in C# so it can actually be tested. The script itself is already broken into various functions so I don't think pulling it apart (assuming you meant into separate files) will make it much easier to work with 🤔. Having said that, we will definitely plan to reduce the size of our bash scripts, and rely on scripts that are testable 👍